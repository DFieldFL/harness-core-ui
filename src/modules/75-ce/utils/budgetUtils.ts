export enum AlertStatus {
  Danger,
  Warning,
  Info
}

interface CostStatusInfo {
  cost: number
  ratio: number
  percentage: number
  status: AlertStatus
}

export interface getAllBudgetCostInfoReturnType {
  actualCostStatus: CostStatusInfo
  forecastedCostStatus: CostStatusInfo
  budgetAmountStatus: CostStatusInfo
}

export const getAllBudgetCostInfo: (
  forecastedCost: number,
  actualCost: number,
  budgetAmount: number
) => getAllBudgetCostInfoReturnType = (forecastedCost, actualCost, budgetAmount) => {
  /**
   * Find out the max cost of all and it will be considered as 100%
   * and the rest of the cost percentages will be calculated based on that
   */

  const baseCost = Math.max(actualCost, forecastedCost, budgetAmount)
  const actualCostRatio = actualCost / baseCost
  const forecastCostRatio = forecastedCost / baseCost
  const budgetAmountRatio = budgetAmount / baseCost

  const actualCostPerWithBudget = (actualCost / budgetAmount) * 100
  const forecastCostPerWithBudget = (forecastedCost / budgetAmount) * 100

  const getAlertStatus: (percentage: number) => AlertStatus = percentage => {
    if (percentage > 100) {
      return AlertStatus.Danger
    }
    if (percentage > 90) {
      return AlertStatus.Warning
    }
    return AlertStatus.Info
  }

  const actualCostStatus: CostStatusInfo = {
    cost: actualCost,
    ratio: actualCostRatio,
    percentage: actualCostPerWithBudget,
    status: getAlertStatus(actualCostPerWithBudget)
  }
  const forecastedCostStatus: CostStatusInfo = {
    cost: forecastedCost,
    ratio: forecastCostRatio,
    percentage: forecastCostPerWithBudget,
    status: getAlertStatus(forecastCostPerWithBudget)
  }
  const budgetAmountStatus: CostStatusInfo = {
    cost: budgetAmount,
    ratio: budgetAmountRatio,
    percentage: 100,
    status: AlertStatus.Info
  }

  return {
    actualCostStatus,
    forecastedCostStatus,
    budgetAmountStatus
  }
}
