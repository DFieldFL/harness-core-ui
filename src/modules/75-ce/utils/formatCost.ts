type FormatCost = (value: number, currency?: string, locale?: string, decimalPoints?: number) => string

const formatCost: FormatCost = (value: number, currency = 'USD', locale = 'en-us', decimalPoints = 2) => {
  if (isNaN(value)) {
    return ''
  }

  return value.toLocaleString(locale, {
    style: 'currency',
    currency: currency,
    maximumFractionDigits: decimalPoints,
    minimumFractionDigits: decimalPoints
  })
}

export default formatCost
