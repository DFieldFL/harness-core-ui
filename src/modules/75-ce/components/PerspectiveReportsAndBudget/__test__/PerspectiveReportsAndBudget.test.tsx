import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import PerspectiveDetailsPage from '../PerspectiveReportsAndBudgets'
import PerspectiveScheduledReportsResponse from './PerspectiveScheduledReportsResponse.json'

jest.mock('services/ce', () => ({
  ...(jest.requireActual('services/ce') as any),
  useGetReportSetting: jest.fn().mockImplementation(() => {
    return { data: PerspectiveScheduledReportsResponse, refetch: jest.fn(), error: null, loading: false }
  })
}))

const params = {
  accountId: 'TEST_ACC',
  perspetiveId: 'perspectiveId',
  perspectiveName: 'sample perspective'
}

describe('test cases for Perspective Reports and Budget page', () => {
  test('should be able to render the reports and budget component', async () => {
    const handlePrevButtonClick = jest.fn().mockImplementationOnce(() => ({}))
    const { container } = render(
      <TestWrapper pathParams={params}>
        <PerspectiveDetailsPage onPrevButtonClick={handlePrevButtonClick} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
