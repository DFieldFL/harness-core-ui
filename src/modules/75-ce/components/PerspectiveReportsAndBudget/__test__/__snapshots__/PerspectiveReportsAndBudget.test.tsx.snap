// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for Perspective Reports and Budget page should be able to render the reports and budget component 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main mainContainer"
  >
    <div
      class="StyledProps--font StyledProps--main innerContainer"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-xxlarge StyledProps--main StyledProps--padding-left-large StyledProps--padding-right-xxlarge StyledProps--padding-bottom-xxlarge StyledProps--padding-top-xxlarge"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
            style="font-size: 16px;"
          >
            ce.perspectives.reports.title
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--padding-bottom-large StyledProps--color StyledProps--color-grey800 StyledProps--font-small"
          >
            ce.perspectives.reports.desc 
          </p>
          <div
            class="tableCtn"
          >
            <table
              role="table"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.reports.reportName
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.reports.frequency
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.reports.recipients
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    Â 
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    Bdj 2
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      Every 5 seconds
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        yo@lo.com
                      </p>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-primary7 StyledProps--font-small"
                            tabindex="0"
                          >
                            (+2)
                          </p>
                        </span>
                      </span>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    Bdj 2
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      Every 5 seconds
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        yo@lo.com
                      </p>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-primary7 StyledProps--font-small"
                            tabindex="0"
                          >
                            (+2)
                          </p>
                        </span>
                      </span>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    Bdj 2
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      Every 5 seconds
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        yo@lo.com
                      </p>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-primary7 StyledProps--font-small"
                            tabindex="0"
                          >
                            (+2)
                          </p>
                        </span>
                      </span>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    Bdj 3
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      Every 2 minutes
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        yo@lo.com
                      </p>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-primary7 StyledProps--font-small"
                            tabindex="0"
                          >
                            (+2)
                          </p>
                        </span>
                      </span>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    sv Report 3
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      At 12:00 AM, on day 1 of the month
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        shubhanshu.verma@nomailplz.com
                      </p>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    sv report 1
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      At 12:00 AM
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        shubhanshu.verma@nomailplz.com
                      </p>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    sv report 2
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      At 12:00 AM, only on Sunday
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                    >
                      <p
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey700 StyledProps--font-small"
                      >
                        shubhanshu.verma@nomailplz.com
                      </p>
                    </div>
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            style="justify-content: flex-end; align-items: center;"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main createBtn Button--with-current-color Button--without-shadow"
              type="submit"
            >
              <span
                class="bp3-button-text"
              >
                ce.perspectives.reports.createNew
              </span>
            </button>
          </div>
        </div>
        <div
          class="StyledProps--font StyledProps--main"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800"
            style="font-size: 16px;"
          >
            ce.perspectives.budgets.title
          </p>
          <p
            class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--padding-bottom-large StyledProps--color StyledProps--color-grey800 StyledProps--font-small"
          >
            ce.perspectives.budgets.desc
          </p>
          <div
            class="tableCtn"
          >
            <table
              role="table"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.budgets.amount
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.budgets.alerts
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.budgets.actualCost
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    ce.perspectives.budgets.lastMonthCost
                  </th>
                  <th
                    colspan="1"
                    role="columnheader"
                  >
                    Â 
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <td
                    role="cell"
                  >
                    5000
                  </td>
                  <td
                    role="cell"
                  >
                    <span>
                      60, 90
                    </span>
                  </td>
                  <td
                    role="cell"
                  >
                    50.64
                  </td>
                  <td
                    role="cell"
                  >
                    1365.06
                  </td>
                  <td
                    role="cell"
                  >
                    <div
                      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
                      style="justify-content: center; align-items: center;"
                    >
                      <span
                        class="bp3-icon StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        data-icon="Edit"
                      >
                        <svg
                          height="14"
                          viewBox="0 0 12 12"
                          width="14"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fill-rule="evenodd"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          >
                            <path
                              d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                              stroke="#555770"
                            />
                            <path
                              d="M6.583 5.146l2.063-2.063"
                              opacity="0.3"
                              stroke="#606170"
                            />
                          </g>
                        </svg>
                      </span>
                      <span
                        class="bp3-icon bp3-icon-trash StyledProps--main icon StyledProps--color StyledProps--color-primary7"
                        icon="trash"
                      >
                        <svg
                          data-icon="trash"
                          height="14"
                          viewBox="0 0 16 16"
                          width="14"
                        >
                          <desc>
                            trash
                          </desc>
                          <path
                            d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
            style="justify-content: flex-end; align-items: center;"
          >
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main createBtn Button--with-current-color Button--without-shadow"
              type="submit"
            >
              <span
                class="bp3-button-text"
              >
                ce.perspectives.budgets.createNew
              </span>
            </button>
          </div>
        </div>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="Layout--horizontal Layout--layout-spacing-large StyledProps--main StyledProps--padding-top-medium"
        >
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
            type="button"
          >
            <span
              class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
              icon="chevron-left"
            >
              <svg
                data-icon="chevron-left"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  chevron-left
                </desc>
                <path
                  d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp3-button-text"
            >
              previous
            </span>
          </button>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              ce.perspectives.save
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
